packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-dynamodb-table-construct/index.ts

```diff
@@ -0,0 +1,377 @@
+      );
+    }
+
+    if (encryptionType === TableEncryption.CUSTOMER_MANAGED && props.replicationRegions) {
+      throw new Error('TableEncryption.CUSTOMER_MANAGED is not supported by DynamoDB Global Tables (where replicationRegions was set)');
+    }
+
+    switch (encryptionType) {
+      case TableEncryption.CUSTOMER_MANAGED:
+        const encryptionKey =
+          props.encryptionKey ??
+          new kms.Key(this, 'Key', {
+            description: `Customer-managed key auto-created for encrypting DynamoDB table at ${this.node.path}`,
+            enableKeyRotation: true,
+          });
+
+        return {
+          sseSpecification: { sseEnabled: true, kmsMasterKeyId: encryptionKey.keyArn, sseType: 'KMS' },
+          encryptionKey,
+        };
+
+      case TableEncryption.AWS_MANAGED:
+        // Not specifying "sseType: 'KMS'" here because it would cause phony changes to existing stacks.
+        return { sseSpecification: { sseEnabled: true } };
+
+      case TableEncryption.DEFAULT:
+        return { sseSpecification: { sseEnabled: false } };
+
+      case undefined:
+        // Not specifying "sseEnabled: false" here because it would cause phony changes to existing stacks.
+        return { sseSpecification: undefined };
+
+      default:
+        throw new Error(`Unexpected 'encryptionType': ${encryptionType}`);
+    }
+  }
+
+  /**
+   * Validate read and write capacity are not specified for on-demand tables (billing mode PAY_PER_REQUEST).
+   *
+   * @param props read and write capacity properties
+   */
+  private validateProvisioning(props: { readCapacity?: number; writeCapacity?: number }): void {
+    if (this.billingMode === BillingMode.PAY_PER_REQUEST) {
+      if (props.readCapacity !== undefined || props.writeCapacity !== undefined) {
+        throw new Error('you cannot provision read and write capacity for a table with PAY_PER_REQUEST billing mode');
+      }
+    }
+  }
+
+  /**
+   * Validate index name to check if a duplicate name already exists.
+   *
+   * @param indexName a name of global or local secondary index
+   */
+  private validateIndexName(indexName: string) {
+    if (this.secondaryIndexSchemas.has(indexName)) {
+      // a duplicate index name causes validation exception, status code 400, while trying to create CFN stack
+      throw new Error(`a duplicate index name, ${indexName}, is not allowed`);
+    }
+  }
+
+  /**
+   * Validate non-key attributes by checking limits within secondary index, which may vary in future.
+   *
+   * @param nonKeyAttributes a list of non-key attribute names
+   */
+  private validateNonKeyAttributes(nonKeyAttributes: string[]) {
+    if (this.nonKeyAttributes.size + nonKeyAttributes.length > 100) {
+      // https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html#limits-secondary-indexes
+      throw new RangeError('a maximum number of nonKeyAttributes across all of secondary indexes is 100');
```

Comment:
Understanding these sorts of limits exist in the table stack. I'm not sure we should be double-mastering these values, otherwise we'll need to stay on top of what Dynamo supports, etc.

Reply:
I think this is for the usage of fast-failing for the incorrect table configs before pushing to the cloud. This is also forked from CDK code. I don't have a strong opinion to remove these code as they are kind of util functions.